### ISSUE TITLE
Repeatable (but not 100% consistent) error in _platform_strcmp

### ISSUE THREAD

**Issue by dgmltn** [dokar3/quickjs-kt#25]
```kotlin
                    val js = """
                        var response = await kml.requestJson('https://icanhazdadjoke.com/');
                        let parsed = JSON.parse(response);
                        parsed.joke;
                    """.trimIndent()

                    val joke = quickJs {
                        define("kml") {
                            asyncFunction("requestJson") { args ->
                                delay(500)
                                """
                                    {"id":"bxXLeqzP7wc","joke":"Americans can't switch from pounds to kilograms overnight. That would cause mass confusion.","status":200}
                                """.trimIndent()
                            }
                        }
                        evaluate<Any?>(js)?.toString()
                    }

```

Causes this error. Not every time but frequently:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000000188e4aa9c, pid=63005, tid=90371
#
# JRE version: OpenJDK Runtime Environment (17.0.7) (build 17.0.7+0-17.0.7b1000.6-10550314)
# Java VM: OpenJDK 64-Bit Server VM (17.0.7+0-17.0.7b1000.6-10550314, mixed mode, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-aarch64)
# Problematic frame:
# C  [libsystem_platform.dylib+0xa9c]  _platform_strcmp+0xc
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
JNI global refs:
JNI global refs: 161, weak refs: 11

JNI global refs memory usage: 2099, weak refs: 833

OOME stack traces (most recent first):
Classloader memory used:
Loader jdk.internal.loader.ClassLoaders$AppClassLoader                                 : 3926K
Loader bootstrap                                                                       : 2694K
Loader jdk.internal.loader.ClassLoaders$PlatformClassLoader                            : 18987B
Loader jdk.internal.reflect.DelegatingClassLoader                                      : 4278B

# An error report file with more information is saved as:
# /Users/dmelton/Developer/doug/kmlauncher/compose-multiplatform/composeApp/hs_err_pid63005.log
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

```

I removed any actual networking, but simulated by that `delay(500)`

**Comment by dokar3** [comment]
Thanks for reporting again. I have no luck reproducing the crash using this code on my devices (tried on Windows and Android).

The `strcmp` call in the report looks suspicious, but the provided problem code doesn't seem to call `strcmp` in JNI, so currently I can't do further debugging on this.

Anyway, the next version has removed some `strcmp` calls, it may help if JNI `strcmp` does cause the crash. The next version may take an hour or more from now to be released, please feel free to try and give feedback when it's ready.

**Comment by dgmltn** [comment]
Ok, no problem, thanks! I think it might be related to the `asyncFunction`. When I changed to `function` and called my coroutine inside `runBlocking` I could not replicate.

**Comment by dokar3** [comment]
That's good to hear, smaller scope to check. And the new version is out, could you test the crash with it and post the full crash report log? If there is something wrong with `asyncFunction`, the new release may produce the same problem.

**Comment by dgmltn** [comment]
Actually it's behaving differently now! I'm no longer getting a native error, I'm getting this error instead. I can understand why `android/os/Build$VERSION` is not found, this error occurs on desktop.

```log
Exception in thread "DefaultDispatcher-worker-4" java.lang.RuntimeException: Exception while trying to handle coroutine exception
	at kotlinx.coroutines.CoroutineExceptionHandlerKt.handlerException(CoroutineExceptionHandler.kt:33)
	at kotlinx.coroutines.internal.CoroutineExceptionHandlerImpl_commonKt.handleUncaughtCoroutineException(CoroutineExceptionHandlerImpl.common.kt:38)
	at kotlinx.coroutines.CoroutineExceptionHandlerKt.handleCoroutineException(CoroutineExceptionHandler.kt:28)
	at kotlinx.coroutines.StandaloneCoroutine.handleJobException(Builders.common.kt:190)
	at kotlinx.coroutines.JobSupport.finalizeFinishingState(JobSupport.kt:228)
	at kotlinx.coroutines.JobSupport.tryMakeCompletingSlowPath(JobSupport.kt:907)
	at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:864)
	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:829)
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:97)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:28)
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:99)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:102)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:585)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:802)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:706)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:693)
	Suppressed: java.lang.Error: InternalError: stack overflow

		at com.dokar.quickjs.QuickJs.getEvaluateResult(Native Method)
		at com.dokar.quickjs.QuickJs.evalAndAwait(QuickJs.jni.kt:217)
		at com.dokar.quickjs.QuickJs.access$evalAndAwait(QuickJs.jni.kt:55)
		at com.dokar.quickjs.QuickJs$evalAndAwait$1.invokeSuspend(QuickJs.jni.kt)
		at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
		at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
		... 4 more
Caused by: java.lang.NoClassDefFoundError: android/os/Build$VERSION
	at kotlinx.coroutines.android.AndroidExceptionPreHandler.handleException(AndroidExceptionPreHandler.kt:43)
	at kotlinx.coroutines.internal.CoroutineExceptionHandlerImpl_commonKt.handleUncaughtCoroutineException(CoroutineExceptionHandlerImpl.common.kt:34)
	... 16 more
Caused by: java.lang.ClassNotFoundException: android.os.Build$VERSION
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	... 18 more
Exception in thread "DefaultDispatcher-worker-4" java.lang.Error: InternalError: stack overflow

	at com.dokar.quickjs.QuickJs.getEvaluateResult(Native Method)
	at com.dokar.quickjs.QuickJs.evalAndAwait(QuickJs.jni.kt:217)
	at com.dokar.quickjs.QuickJs.access$evalAndAwait(QuickJs.jni.kt:55)
	at com.dokar.quickjs.QuickJs$evalAndAwait$1.invokeSuspend(QuickJs.jni.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:585)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:802)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:706)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:693)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [kotlinx.coroutines.UndispatchedMarker@551cf53, androidx.compose.runtime.BroadcastFrameClock@15ef4628, StandaloneCoroutine{Cancelling}@2faefea5, Dispatchers.Default]

```


on android, this is the error that occurs:
```log
FATAL EXCEPTION: DefaultDispatcher-worker-14
Process: com.dgmltn.kmlauncher, PID: 22489
java.lang.Error: InternalError: stack overflow

    at com.dokar.quickjs.QuickJs.getEvaluateResult(Native Method)
    at com.dokar.quickjs.QuickJs.evalAndAwait(QuickJs.jni.kt:217)
    at com.dokar.quickjs.QuickJs.access$evalAndAwait(QuickJs.jni.kt:55)
    at com.dokar.quickjs.QuickJs$evalAndAwait$1.invokeSuspend(Unknown Source:15)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
    at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:585)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:802)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:706)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:693)
    Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@f0e68c4, Dispatchers.Default]

```

Hope this helps!

**Comment by dokar3** [comment]
Interesting, it looks like QuickJS threw a stack overflow error. Does this only happen if using `asyncFunction`? I have seen stack overflow errors before but it was not related to async or suspend calls, after I increased the `maxStackSize` QuickJs worked again. Can you increase the `maxStackSize` to see if it's still there?

```kotlin
quickJs {
    maxStackSize = 5 * 1024 * 1024L
}
```

**Comment by dgmltn** [comment]
Yes, it seems to only happen when using `asyncFunction`, and bumping the `maxStackSize` didn't seem to help, I still see that stack overflow.

**Comment by dokar3** [comment]
Okay, thanks for the information. Will look into this later.

**Comment by dokar3** [comment]
I spent some hours looking into this today, but, unfortunately, I still have no luck reproducing this stack overflow error.

I requested `https://icanhazdadjoke.com/` in a suspend function using `ktor`, then ran it on my PC (Windows X64), my phone (Android arm64-v8a), and an emulator (Android x84_64). However, nothing happened, I got all JSON responses, even though I changed the code to fetch 3 responses at once using `Promise.all()`.

I think this is a device/OS related issue and I can't dig deeper into this issue now. Would be glad if you find something new about this error or about reproducing it.
